mobile/library/common/extensions/filters/http/network_configuration/filter.cc

```diff
@@ -64,19 +96,70 @@ NetworkConfigurationFilter::decodeHeaders(Http::RequestHeaderMap& request_header
     return Http::FilterHeadersStatus::Continue;
   }
 
-  // If there is no proxy configured, continue.
+  // TODO(Augustyniak): Update Android proxy resolution to use API extension registry. As of now,
+  // it's only iOS that uses that code path.
+  const auto proxy_resolver =
+      static_cast<envoy_proxy_resolver*>(Api::External::retrieveApiSafe("envoy_proxy_resolver"));
+  if (proxy_resolver != nullptr) {
+    return resolveProxy(request_headers, proxy_resolver);
+  }
+
+  // TODO(Augustyniak): Migrate Android so that it uses API registry instead of calling
```

Comment:
Android was originally implemented using a synchronous API for getting a list of proxies. This does not work for iOS as PAC proxy resolution is async. That's why iOS implementation of proxies introduces a new async mechanism for resolving proxies. 

The idea I had is to:
- introduce new async mechanism for proxies on iOS while continuing to use the old mechanism for proxies on Android
- eventually migrate Android proxies implementation to new async mechanism